client/connection/manager.go

```diff
@@ -43,11 +43,18 @@ func NewManager(mysteriumClient server.Client, dialogEstablisherCreator DialogEs
 	}
 }
 
-func (manager *connectionManager) Connect(consumerID, providerID identity.Identity) error {
+func (manager *connectionManager) Connect(consumerID, providerID identity.Identity) (err error) {
 	if manager.status.State != NotConnected {
 		return ErrAlreadyExists
 	}
 
+	manager.status = statusConnecting()
+	defer func() {
+		if err != nil {
+			manager.status = statusNotConnected()
```

Comment:
Would be nice this assert this status in tests

Reply:
It was already covered, but I cleaned up the tests

Reply:
@tadovas @Waldz

Reply:
Anyway missing test about status Connecting

Reply:
Added test case. @Waldz  up to you now.